extensions/ql-vscode/src/mocks/recorder.ts

```diff
@@ -0,0 +1,176 @@
+import * as fs from 'fs-extra';
+import * as path from 'path';
+
+import { MockedRequest } from 'msw';
+import { SetupServerApi } from 'msw/node';
+import { IsomorphicResponse } from '@mswjs/interceptors';
+
+import { DisposableObject } from '../pure/disposable-object';
+
+import { GitHubApiRequest, RequestKind } from './gh-api-request';
+
+export class Recorder extends DisposableObject {
+  private readonly allRequests = new Map<string, MockedRequest>();
+  private currentRecordedScenario: GitHubApiRequest[] = [];
+
+  private _recording = false;
```

Comment:
This is a personal preference thing so feel free to ignore (although I'd like it to be a linting rule at some point): I think we should avoid prefixing variables with `_`. I don't think it really brings anything other than making people think the variable is special in some way.

Reply:
I agree with you that in general we shouldn't prefix these with `_` but in this case it's done because we want the variable to be read-only outside of the class, but read-write inside of the class. I think choosing two different names for the public getter and private variable would make things even more confusing, hence why I chose to prefix it like this.

Reply:
But you have provided two different names - `isRecording` and `_recording`. Apologies if I'm missing something obvious!

Reply:
Sorry about that! I renamed the method but didn't rename the variable. I've now aligned them so they match.